cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java

```diff
@@ -516,21 +522,63 @@ protected String maybeAddMonitorIntervalToExternalLogging(String data) {
     /**
      * Generates a metrics and logging ConfigMap according to configured defaults.
      *
-     * @param externalConfigMap The ConfigMap used if Logging is an instance of ExternalLogging
+     * @param externalLoggingConfigMap The ConfigMap used if Logging is an instance of ExternalLogging
+     * @param externalMetricsConfigMap The ConfigMap used if Metrics is an instance of ExternalMetrics
      * @return The generated ConfigMap.
      */
-    public ConfigMap generateMetricsAndLogConfigMap(ConfigMap externalConfigMap) {
+    public ConfigMap generateMetricsAndLogConfigMap(ConfigMap externalLoggingConfigMap, ConfigMap externalMetricsConfigMap) {
         Map<String, String> data = new HashMap<>(2);
-        data.put(getAncillaryConfigMapKeyLogConfig(), parseLogging(getLogging(), externalConfigMap));
-        if (isMetricsEnabled() && getMetricsConfig() != null) {
+        data.put(getAncillaryConfigMapKeyLogConfig(), parseLogging(getLogging(), externalLoggingConfigMap));
+        if (getMetricsConfigInCm() != null || (isMetricsEnabled() && getMetricsConfig() != null)) {
```

Comment:
I was wondering if we really still need the `isMetricsEnabled` when everything depends on the `metrics` field. If the field is not specified it just means no metrics. Or there is something missing  why it's still needed?

Reply:
AFAICS the purpose of this field is to be written into the envvar, what causes RU if we want to un/deploy metrics.